search/includes/classes/class-versioning.php

```diff
@@ -40,17 +61,23 @@ public function __construct() {
 		// the Queue to take over EP's queue, at which point we don't need to insert them here, as they are handled during Queue::queue_object())
 		add_filter( 'pre_ep_index_sync_queue', [ $this, 'filter__pre_ep_index_sync_queue' ], 100, 3 );
 
-		add_action( 'plugins_loaded', [ $this, 'action__plugins_loaded' ] );
+		add_action( 'init', [ $this, 'action__elasticpress_loaded' ], PHP_INT_MAX );
```

Comment:
Check out this change. Seems the deletion only ever worked for post type as the features are enabled on [init](https://github.com/10up/ElasticPress/blob/a2545cbb96740694ff9c5067a5f634fafef16932/includes/classes/Features.php#L36)

Which happens after plugins_loaded.

Reply:
Oh good catch - what do you think about splitting this out into a separate PR, since it's mostly unrelated? We can get this merged/deployed independently.

Reply:
I created this PR: https://github.com/Automattic/vip-go-mu-plugins/pull/1903

However I need that change here as well, otherwise you can't test partial self-healing (e.g. only of post, but leaving users intact).

We can then squash this and rebase when the 1903 gets merged.